app/models/subscription.rb

```diff
@@ -2,19 +2,38 @@ class Subscription < ApplicationRecord
   belongs_to :user
   belongs_to :plan
   has_many :transactions
+  has_many :usages, dependent: :destroy
 
   validates :billing_day, presence: true, inclusion: 1..28
 
   before_validation :set_billing_day
   after_save :subscription_transaction
 
+  def charge_fee
+    subscription_transaction
+  end
+
+  def create_usage(usage_params)
+    usages.create(usage_params)
+  end
```

Comment:
do we still need this method?

Reply:
We are calling this method to create a usage of a Subscription.

Reply:
as discussed, remove this method and create usage as advised above.